app/main.py

```diff
@@ -1,8 +1,18 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    instances = [Person(person["name"], person["age"]) for person in people]
+    for person in people:
+        current_person = Person.people[person["name"]]
+        if person.get("wife"):
+            current_person.wife = Person.people[person["wife"]]
+        if person.get("husband"):
+            current_person.husband = Person.people[person["husband"]]
```

Comment:
You could use walrus operator to make code more efficient and avoid double value retrieval from dict
```suggestion
        if wife_name := person.get("wife"):
            current_person.wife = Person.people[wife_name]
        if husband_name := person.get("husband"):
            current_person.husband = Person.people[husband_name]
```